src/app/results/results.component.ts

```diff
@@ -213,6 +210,21 @@ export class ResultsComponent implements OnInit {
       this.presentPage = Math.abs(query['start'] / urldata.rows) + 1;
       let querydata = Object.assign({}, urldata);
       this.store.dispatch(new queryactions.QueryServerAction(querydata));
+        this.getJsonService.getJSON().subscribe(res => { this.newsResponse = [];
+          for (let i = 0; i < res.newsOrgs.length; i++) {
+          this.getNewsService.getSearchResults(querydata, res.newsOrgs[i].provider).subscribe(
+            response => {
+                if (typeof response.channels[0].items[0] !== 'undefined') {
```

Comment:
Why using typeof if you are comparing it to check if its undefined or not. It will be string only.

Reply:
![image](https://user-images.githubusercontent.com/25537606/42024472-e773692e-7adf-11e8-9eea-88a3f109053c.png)
Both are correct I think!

Reply:
Any advantage of using string?

Reply:
I am not saying to use string.
I am saying it could be as simple: ```if (response.channels[0].items[0] !== 'undefined')```?

Reply:
I think we should make it `if (response.channels[0].items[0] !== undefined)`